java/ql/src/experimental/Security/CWE/CWE-295/HostnameValidation.qll

```diff
@@ -0,0 +1,140 @@
+import java
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.security.Encryption
+
+/** A method that overrides the `javax.net.ssl.HostnameVerifier.verify` method. */
+class OverridenVerifyMethod extends HostnameVerifierVerify {
+  OverridenVerifyMethod() { this instanceof HostnameVerifierVerify }
+
+  /**
+   * The `hostname` parameter of this method.
+   */
+  Parameter getHostnameParameter() { result = getParameter(0) }
+
+  /**
+   * The `session` parameter of this method.
+   */
+  Parameter getSessionParameter() { result = getParameter(1) }
+}
+
+/** A return statement that returns `true`. */
+class TrueReturnStmt extends ReturnStmt {
+  TrueReturnStmt() { getResult().(BooleanLiteral).getBooleanValue() = true }
```

Comment:
Use `(CompileTimeConstantExpr).getBooleanValue()` unless `BooleanLiteral` extends `CompileTimeConstantExpr`.

Reply:
Thank you for that suggestion.
I've applied it!